src/codegen/x86_64/arch_x86_64.c

```diff
@@ -1091,56 +1203,93 @@ static void emit_instruction(CodegenContext *context, IRInstruction *inst) {
     break;
 
   case IR_LOAD:
+    // TODO: Handle size of type and stuff
     /// Load from a static variable.
     if (inst->operand->kind == IR_STATIC_REF) {
-      femit(context, I_MOV, NAME_TO_REGISTER, REG_RIP, inst->operand->static_ref->name.data,
-            inst->result);
+      enum RegSize size = -1;
+      // TODO: Should this array to pointer decay happen here? Or higher up in codegen?
+      // TODO: type_sizeof(t_pointer) or something to load a pointer sized thing.
+      //       WE SHOULD NOT USE t_integer here!!
+      if (inst->operand->type->kind == TYPE_ARRAY) size = regsize_from_bytes(type_sizeof(t_integer));
+      else size = regsize_from_bytes(type_sizeof(inst->operand->type));
```

Comment:
That makes sense, yeah. Basically remove the temporary from `a -> tmp -> b` if we can just move `a -> b` directly.

So a load of an array type, in the x86_64 backend, should probably use registers and multiple `mov`s to get the job done. Not only that, there will probably be a lowering step where stack allocation is inserted for load of array, right?

Reply:
> So a load of an array type, in the x86_64 backend, should probably use registers and multiple `mov`s to get the job done.

For small arrays, yeah. For huge arrays I’d generate a call to `memcpy()` or `REP MOVSB` or similar. But that’s not really that important for now.

> Not only that, there will probably be a lowering step where stack allocation is inserted for load of array, right?

That may or may not be necessary. I suspect the only way to figure this out is by looking at specific cases and going from there. We have to figure this out at some point since we’re gonna have the same problem w/ structs and really anything that doesn’t fit in a single register...

Reply:
> A ‘load’ of an array might require allocating a copy on the stack I think.

There are actually two ways we need to "load an array"

1. `foo[2]` :: we need to load **the address of** `foo`, add immediate `2` multiplied by sizeof base type, and then that gets us the proper address
2. `a : foo[2] = b` where b is also a foo[2], we need to actually load entire array